packages/x-components/src/x-modules/queries-preview/components/query-preview.vue

```diff
@@ -222,21 +222,33 @@
           }
         }
       );
-      this.emitQueryPreviewRequestUpdated(this.queryPreviewRequest);
+
+      const previewItemQuery = this.queryPreviewInfo.query;
+      const isSavedQuery = !!this.previewResults[previewItemQuery];
+
+      // If the query has been saved it will emit load instead of the emitting the updated request.
+      if (isSavedQuery && this.persistInCache) {
+        if (this.previewResults[previewItemQuery].status !== 'error') {
+          this.$emit('load', this.queryPreviewInfo.query);
+        } else {
+          this.emitQueryPreviewRequestUpdated(this.queryPreviewRequest);
+        }
+      } else {
+        this.emitQueryPreviewRequestUpdated(this.queryPreviewRequest);
+      }
     }
 
     /**
      * Cancels the (remaining) requests when the component is destroyed
      * via the `debounce.cancel()` method.
-     * If the prop 'clearOnDestroy' is set to true, it also removes the QueryPreview
+     * If the prop 'persistInCache' is set to false, it also removes the QueryPreview
      * from the state when the component is destroyed.
      *
      * @internal
      */
     protected beforeDestroy(): void {
       this.emitQueryPreviewRequestUpdated.cancel();
-
-      if (this.clearOnDestroy) {
+      if (!this.persistInCache) {
         this.$x.emit('QueryPreviewUnmountedHook', this.queryPreviewInfo.query, {
```

Comment:
I hate this event name xD. Can you change it to `NonCacheableQueryPreviewUnmounted`? 🙏